apps/src/code-studio/pd/application_dashboard/detail_view.jsx

```diff
@@ -26,48 +21,37 @@ export default class DetailView extends React.Component {
     })
   };
 
-  state = {
-    loading: true
+  static contextTypes = {
+    router: PropTypes.object.isRequired
   };
 
-  componentWillMount() {
-    this.load();
-  }
-
+  handleApplicationLoaded = applicationData => {
+    const {course, application_type} = applicationData;
+    const applicationId = this.props.params.applicationId;
```

Comment:
It's for readability. Nothing is ever really needed, is it?

Line [35](https://github.com/code-dot-org/code-dot-org/pull/20520/files#diff-b7fe47c0f7ee4d7192e9a13c3819e605R35) is now:
```jsx
 this.context.router.replace(`/${course}_${application_type.toLowerCase()}s/${applicationId}`);
``` 

Without the constant, it would be:
```jsx
 this.context.router.replace(`/${course}_${application_type.toLowerCase()}s/${this.props.params.applicationId}`);
```

Since that line is already long, I thought it more readable the first way. I'm happy to change it back, or move the const declaration inside that conditional.